### PR TITLE
Implement vaccine support chart by age group

### PR DESCRIPTION
**Pull request by VWSCoronaDashboard8**
## Summary

For this task I created the MultiLineChart as a fork from LineChart, because there were some incompatibilities. Namely the tooltip needed to be rendered on one side and flip when hitting bounds. Our custom tooltip doesn't have logic for this.

I have forked the ChartAxes because this chart renderers sub grid lines and has specific tick value settings. 

Also, the way the tooltip gets its data handed in this chart is, I believe, an improvement over the LineChart approach. One part of it (the active hovered data property) isn't yet implemented but I've left the intention there as a note.  It is somewhat similar to how the tooltip in the stacked-chart component works.

In other words, this chart becomes yet another study subject in our quest to make reusable chart components.

image

### CODE REVIEW

**Review by VWSCoronaDashboard1 — APPROVED** [review]

File: packages/app/schema/nl/vaccine_support.json [hunk 1]
```diff
@@ -20,10 +20,22 @@
       "title": "nl_vaccine_support_value",
       "type": "object",
       "properties": {
-        "percentage_in_favor": {
+        "percentage_total": {
           "type": "number"
         },
-        "percentage_already_vaccinated": {
+        "percentage_70_plus": {
```
Comment (VWSCoronaDashboard4) [inline packages/app/schema/nl/vaccine_support.json:26]: these properties are not required?

**Review by VWSCoronaDashboard4 — APPROVED** [review]

File: packages/app/schema/nl/vaccine_support.json (hunk 1 above)
Comment (VWSCoronaDashboard8) [reply packages/app/schema/nl/vaccine_support.json:26]: Good catch thanks

**Review by VWSCoronaDashboard1 — APPROVED** [review]

File: packages/app/src/components-styled/line-chart/components/marker.tsx [hunk 2]
```diff
@@ -126,18 +126,18 @@ export function Marker<T extends TimestampedValue>(props: MarkerProps<T>) {
         <LineContainer
           style={{
             top: 'calc(100% + 5px)',
-            left: data[0].x,
+            left: data[0].x - 1,
```
[the remark this answers is not in the document]
Comment (VWSCoronaDashboard8) [reply packages/app/src/components-styled/line-chart/components/marker.tsx:129]: I wouldn't know a name for this, other than "it improves the alignment". But I can add a comment.

File: packages/app/src/components-styled/multi-line-chart/components/chart-axes.tsx [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+  label: string;
+};
+
+export type ChartScales = {
+  xScale: ScaleTime<number, number>;
+  yScale: ScaleLinear<number, number>;
+};
+
+type ChartAxesProps = {
+  benchmark?: Benchmark;
```
Comment (VWSCoronaDashboard4) [inline packages/app/src/components-styled/multi-line-chart/components/chart-axes.tsx:55]: Order the optional props last?

File: packages/app/src/components-styled/multi-line-chart/components/chart-axes.tsx [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+          tickLabelProps={() => ({
+            dx: -25,
+            fill: defaultColors.axisLabels,
+            fontSize: 12,
+          })}
+          hideTicks
+        />
+        <AxisLeft
+          scale={yScale}
+          // numTicks={NUM_TICKS}
```
Comment (VWSCoronaDashboard4) [inline packages/app/src/components-styled/multi-line-chart/components/chart-axes.tsx:151]: nted out code?

File: packages/app/src/pages/landelijk/vaccinaties.tsx [hunk 5]
```diff
@@ -263,36 +262,113 @@ const VaccinationPage: FCWithLayout<typeof getStaticProps> = ({
...
           }}
         >
+          <section>
+            <KpiValue
+              percentage={data.vaccine_support.last_value.percentage_average}
+            />
+            <Text mt={0}>{text.grafiek_draagvlak.kpi_omschrijving}</Text>
+          </section>
+
+          {/* <Text mt={0}>{text.grafiek_draagvlak.omschrijving}</Text> */}
```
Comment (VWSCoronaDashboard4) [inline packages/app/src/pages/landelijk/vaccinaties.tsx:272]: commented out code?

File: packages/app/src/pages/landelijk/vaccinaties.tsx [hunk 6]
```diff
@@ -327,3 +403,36 @@ const VaccinationPage: FCWithLayout<typeof getStaticProps> = ({
 VaccinationPage.getLayout = getNationalLayout;
 
 export default VaccinationPage;
+
+const TooltipList = styled.ol`
+  margin: 0;
+  padding: 0;
+  list-style: 'none';
```
Comment (VWSCoronaDashboard4) [inline packages/app/src/pages/landelijk/vaccinaties.tsx:410]: Shouldn't it be without the quotes, since you are in the interpolation way of passing styles?

```suggestion
  list-style: none;
```

File: packages/app/src/pages/landelijk/vaccinaties.tsx [hunk 7]
```diff
@@ -263,36 +262,113 @@ const VaccinationPage: FCWithLayout<typeof getStaticProps> = ({
...
+                  {
+                    metricProperty: 'percentage_70_plus',
+                    label: '70+ jaar',
+                    color: '#C252D4',
+                    legendShape: 'square',
+                    areaFillOpacity: 0,
+                  },
+                ]}
+                /**
+                 * @TODO The tooltip formatting is getting a bit out of hand here. I
```
Comment (VWSCoronaDashboard4) [inline packages/app/src/pages/landelijk/vaccinaties.tsx:321]: good to take into account when we are unifying the tooltip approach!

File: packages/app/src/components-styled/multi-line-chart/components/chart-axes.tsx [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+          width={bounds.width}
+          numTicks={5}
+          tickValues={yTickValues}
+          stroke={defaultColors.gridMain}
+        />
+        <AxisBottom
+          scale={xScale}
+          tickValues={xScale.domain()}
+          tickFormat={formatXAxis as AnyTickFormatter}
+          top={bounds.height}
```
Comment (VWSCoronaDashboard4) [inline packages/app/src/components-styled/multi-line-chart/components/chart-axes.tsx:140]: Shouldn't this be `padding.top + bound.height` (perhaps adjusted by an offset? What is going to happen if you pass a different padding.top?

**Review by VWSCoronaDashboard4 — COMMENTED** [review]
Looking good, code wise. Some nitpicks/questions

File: packages/app/src/components-styled/multi-line-chart/components/chart-axes.tsx (hunk 3 above)
Comment (VWSCoronaDashboard8) [reply packages/app/src/components-styled/multi-line-chart/components/chart-axes.tsx:55]: This component was forked, but yeah I agree it's nicer to group them last

File: packages/app/src/pages/landelijk/vaccinaties.tsx (hunk 6 above)
Comment (VWSCoronaDashboard8) [reply packages/app/src/pages/landelijk/vaccinaties.tsx:410]: Somehow this works too though :)

File: packages/app/src/components-styled/multi-line-chart/components/chart-axes.tsx (hunk 8 above)
Comment (VWSCoronaDashboard8) [reply packages/app/src/components-styled/multi-line-chart/components/chart-axes.tsx:140]: I don't know. Again, this was forked, and I didn't write the original code. The whole abstraction needs to be removed anyway, because we already concluded (I think) that it's hurting more than helping. So I'm going to leave this as-is

**Review by VWSCoronaDashboard10 — APPROVED** [review]

File: packages/app/src/components-styled/multi-line-chart/multi-line-chart.tsx [hunk 9]
```diff
@@ -0,0 +1,539 @@
...
+          tooltipData: {
+            /**
+             * Ideally I think we would pass the original value + the key that
+             * this hover point belongs to. Similar to how the stacked-chart
+             * hover works. But in order to do so I think we need to use
+             * different hover logic, and possibly use mouse callbacks on the
+             * trends individually.
+             */
+            value: hoverPoints[0].data,
+            key: '__todo_figure_out_what_key_this_point_belongs_to' as keyof T,
```
[the remark this answers is not in the document]
Comment (VWSCoronaDashboard8) [reply packages/app/src/components-styled/multi-line-chart/multi-line-chart.tsx:293]: Yes but only a todo as in a reminder that I wanted to reach this type of interface but didn't succeed yet. It's something to look at when we start consolidating charts and their format tooltip signature I think. This is what I was talking about in the description.

File: packages/app/src/components-styled/multi-line-chart/multi-line-chart.tsx [hunk 10]
```diff
@@ -0,0 +1,539 @@
...
+          return trendValue
+            ? {
+                data: trendValue,
+                color: linesConfig[index].color,
+              }
+            : undefined;
+        })
+        .filter(isDefined)
+        .map<HoverPoint<T>>(
+          ({ data, color }: { data: any; color?: string }) => {
```
[the remark this answers is not in the document]
Comment (VWSCoronaDashboard8) [reply packages/app/src/components-styled/multi-line-chart/multi-line-chart.tsx:347]: This code is forked (copy/pasted/adjusted) from line-chart. Some of the code in here is not to stay and outside of this PR AFAIC

File: packages/app/src/components-styled/multi-line-chart/multi-line-chart.tsx [hunk 11]
```diff
@@ -0,0 +1,539 @@
...
+      return +date - +d0.__date > +d1.__date - +date ? d1 : d0;
+    },
+    [padding]
+  );
+
+  const distance = (point1: HoverPoint<TimestampedValue>, point2: Point) => {
+    const x = point2.x - point1.x;
+    const y = point2.y - point1.y;
+    return Math.sqrt(x * x + y * y);
+  };
```
[the remark this answers is not in the document]
Comment (VWSCoronaDashboard8) [reply packages/app/src/components-styled/multi-line-chart/multi-line-chart.tsx:271]: Since this was forked I'll leave it as is for now.
